DllImportGenerator/DllImportGenerator/StubCodeGenerator.cs

```diff
@@ -16,20 +17,76 @@ private StubCodeGenerator(Stage stage)
             CurrentStage = stage;
         }
 
+        public override bool PinningSupported => true;
+
+        public override bool StackSpaceUsable => true;
+
+        /// <summary>
+        /// Identifier for managed return value
+        /// </summary>
+        public const string ReturnIdentifier = "__retVal";
+
+        /// <summary>
+        /// Identifier for native return value
+        /// </summary>
+        /// <remarks>Same as the managed identifier by default</remarks>
+        public string ReturnNativeIdentifier { get; private set; } = ReturnIdentifier;
+
+        private const string InvokeReturnIdentifier = "__invokeRetVal";
+
+        /// <summary>
+        /// Generate an identifier for the native return value and update the context with the new value
+        /// </summary>
+        /// <returns>Identifier for the native return value</returns>
+        public void GenerateReturnNativeIdentifier()
+        {
+            if (CurrentStage != Stage.Setup)
+                throw new InvalidOperationException();
+
+            // Update the native identifier for the return value
+            ReturnNativeIdentifier = $"{ReturnIdentifier}{GeneratedNativeIdentifierSuffix}";
+        }
+
+        public override (string managed, string native) GetIdentifiers(TypePositionInfo info)
+        {
+            if (info.IsManagedReturnPosition && !info.IsNativeReturnPosition)
+            {
+                return (ReturnIdentifier, ReturnNativeIdentifier);
+            }
+            else if (!info.IsManagedReturnPosition && info.IsNativeReturnPosition)
+            {
+                return (InvokeReturnIdentifier, InvokeReturnIdentifier);
+            }
+            else if (info.IsManagedReturnPosition && info.IsNativeReturnPosition)
+            {
+                return (ReturnIdentifier, ReturnNativeIdentifier);
+            }
+            else
+            {
+                return base.GetIdentifiers(info);
```

Comment:
Let's assert and state the intuition for why the not managed and not native return condition.